### PR TITLE
support atomic transactions

### PR DESCRIPTION
**Pull request by krlosMata**
This PR does the following:
- implements method `buildTxCompressedDataV2`
- implements logic to support atomic transactions
- add tests for `buildTxCompressedDataV2` & atomic transactions

### CODE REVIEW

**Review by AlbertoElias — APPROVED** [review]

File: src/tx-utils.js [hunk 1]
```diff
@@ -357,11 +405,39 @@ function buildTransactionHashMessage (encodedTransaction) {
...
+ * Prepares a transaction to be ready to be sent to a Coordinator (without encodedTx)
+ * @param {Object} transaction - ethAddress and babyPubKey together
+ * @param {String} transaction.from - The account index that's sending the transaction e.g hez:DAI:4444
+ * @param {String} transaction.to - The account index or Hermez address of the receiver e.g hez:DAI:2156. If it's an Exit, set to a falseable value
+ * @param {HermezCompressedAmount} transaction.amount - The amount being sent as a HermezCompressedAmount
+ * @param {Number} transaction.fee - The amount of tokens to be sent as a fee to the Coordinator
+ * @param {Number} transaction.nonce - The current nonce of the sender's token account (optional)
+ * @param {String} bjj - The compressed BabyJubJub in hexadecimal format of the transaction sender
+ * @param {Object} token - The token information object as returned from the Coordinator.
+ * @return {Object} - Contains `transaction` and `encodedTransaction`. `transaction` is the object almost ready to be sent to the Coordinator. `encodedTransaction` is needed to sign the `transaction`
```
Comment (AlbertoElias) [inline src/tx-utils.js:438]: No longer contains encodedTransaction

File: tests/helpers/helpers.js [hunk 2]
```diff
@@ -93,3 +95,36 @@ export async function printBalances (accounts, token) {
...
+  const generateTxs = []
+  const atomicTxs = []
+  if (txs.length !== wallets.length || txs.length !== tokens.length) { throw new Error('Invalid atomic group') }
+  for (let i = 0; i < txs.length; i++) {
+    const tx = await computeL2Transaction(txs[i], wallets[i], tokens[i])
+    generateTxs.push(tx)
+  }
+  for (let j = 0; j < generateTxs.length; j++) {
+    let atomicTx
+    if (j !== generateTxs.length - 1) {
```
Comment (AlbertoElias) [inline tests/helpers/helpers.js:122]: why is this? might be good to add a comment

**Review by AlbertoElias — CHANGES_REQUESTED** [review]
A few small comments

File: src/atomic-utils.js [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+import { TX_ID_BYTES } from './constants'
+
+/**
+ * Determines if the transaction has a linked transaction
+ * @param {Object} transaction - Transaction object returned by generateL2Transaction
+ * @returns {Boolean} True if a transaction is linked, false otherwise
+ */
+function hasLinkedTransaction (transaction) {
+  if (
+    (transaction.requestFromAccountIndex !== null && transaction.requestFromAccountIndex !== undefined) ||
```
[the remark this answers is not in the document]
Comment (krlosMata) [reply src/atomic-utils.js:15]: oh! thanks for sharing!
Although I think here does not fit perfectly

**Review by AlbertoElias — APPROVED** [review]

**Pull request merged**